src/ui/components/UploadScreen/UploadScreen.tsx

```diff
@@ -56,6 +57,23 @@ function DeletedScreen({ url }: { url: string }) {
   );
 }
 
+function SoftLimitWarning() {
+  return (
+    <div className="flex flex-row space-x-2 border rounded-lg p-4 border-yellow-400 bg-yellow-50">
+      <div className="flex flex-col space-y-2">
+        <div className="flex flex-row text-yellow-800 font-medium text-xl items-center">
+          <MaterialIcon className="text-yellow-800">warning_amber</MaterialIcon>
+          <div>This replay is over 2 minutes</div>
+        </div>
+        <div>
+          For the best debugging experience, we recommend keeping your replays brief. In the future,
+          please try to keep your Replays under 2 minutes.
+        </div>
```

Comment:
Perhaps something simpler like this? 

---

**2+ minutes**

We recommend keeping replays below 2 minutes for the best debugging experience. _Learn More_